Review: do not merge this

Nariun opened the pull request.

ThomassenCathleia committed:
Chore: first commit. This commit has the foundations of the project's
model layer.

ThomassenCathleia committed:
Renamed MenuItem to OrderItem

Changed the menuItems type in Menu to Product

Commit:
Added a receipt class

ThomassenCathleia committed:
See MCJ_06 in McJavaKitchen for full details

Summary:
- added constructor to every class
- implemented all methods with business logic

ThomassenCathleia committed:
Merge remote-tracking branch 'origin/MCJ_06' into MCJ_06

ThomassenCathleia committed:
Merge pull request #1 from lenshuygh/MCJ_06

Mcj 06

Nariun commented on src/be/mcjava/model/CustomerOrder.java:
```diff
@@ -0,0 +1,89 @@
+        this(null,name,telephoneNumber,itemsToOrder);
+    }
+    
+    public CustomerOrder(Long id, String name, String telephoneNumber, List<OrderItem> itemsToOrder) {
```
4 parameters zijn te veel.
Gebruik een Builder.

ThomassenCathleia replied:
Switched to builder.

Nariun commented on src/be/mcjava/model/CustomerOrder.java:
```diff
@@ -0,0 +1,89 @@
+    }
+    
+    public void setFinished(boolean finished) {
+        boolean actuallyFinished = getItemsToOrder().stream()
```
Degene die de methode oproept weet niet of het nu finished is of niet.
Indien er extra logica is voorzien moet er of wel javadoc zijn die het uitlegt of wel moet de naam van de methode duidelijk zijn.

ThomassenCathleia replied:
Zou nog een exception moeten gooien als het niet lukt.

ThomassenCathleia replied:
Code adjusted.

Commit:
Replaced constructors with builders

Renamed finished to ordered (naming conflict with McJavaKitchen).
Removed the setter logic as well.

Commit:
Merged the two projects and repositories

Fixed the class documentation (where broken)

Added a variable to CustomerOrder, named finishedCooking in order to
avoid naming collisions with finished. finished has been renamed to
ordered.
Ordered now simply takes a boolean value with no special logic.
finishedCooking checks if all items are cooked.

OrderItem and Ingredient now take a prime number multiplication on top
of the hashcode.

ThomassenCathleia committed:
Merge pull request #3 from lenshuygh/MCJ_06_fix

Replaced constructors with builders

ThomassenCathleia committed:
Removed unused import

ThomassenCathleia committed:
Merge pull request #4 from lenshuygh/MCJ_023

Mcj 023
